pkg/scheduler/watchers/membercluster/controller_integration_test.go

```diff
@@ -444,6 +444,41 @@ var _ = Describe("scheduler member cluster source controller", Serial, Ordered,
 		})
 	})
 
+	Context("ready cluster has taint added & removed", func() {
+		BeforeAll(func() {
+			Consistently(noKeyEnqueuedActual, consistentlyDuration, consistentlyInterval).Should(Succeed(), "Workqueue is not empty")
+
+			// Retrieve the cluster.
+			memberCluster := &clusterv1beta1.MemberCluster{}
+			Expect(hubClient.Get(ctx, types.NamespacedName{Name: clusterName1}, memberCluster)).To(Succeed(), "Failed to get member cluster")
+
+			// Add taint to cluster.
+			memberCluster.Spec.Taints = append(memberCluster.Spec.Taints, clusterv1beta1.Taint{
+				Key:    "key1",
+				Value:  "value1",
+				Effect: corev1.TaintEffectNoSchedule,
+			})
+			Expect(hubClient.Update(ctx, memberCluster)).Should(Succeed(), "Failed to update member cluster taints")
+			Consistently(noKeyEnqueuedActual, consistentlyDuration, consistentlyInterval).Should(Succeed(), "Workqueue is not empty")
+
+			// Retrieve the cluster.
+			Expect(hubClient.Get(ctx, types.NamespacedName{Name: clusterName1}, memberCluster)).To(Succeed(), "Failed to get member cluster")
+
+			// Remove taint from cluster.
+			memberCluster.Spec.Taints = nil
+			Expect(hubClient.Update(ctx, memberCluster)).Should(Succeed(), "Failed to update member cluster taints")
+		})
+
+		It("should enqueue CRPs for cluster removed taint (case 1c)", func() {
+			Eventually(qualifiedKeysEnqueuedActual(), eventuallyDuration, eventuallyInterval).Should(Succeed(), "Keys are not enqueued as expected")
```

Comment:
```suggestion
			Eventually(qualifiedKeysEnqueuedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Keys are not enqueued as expected")
```

You should not call the func, otherwise, it will populate the result. so the func will be called just one time.

You need to pass in the func instead.